### PR TITLE
improved `prime_check` in math modules

### PR DESCRIPTION
**Pull request by Lakshmikanth2001**
### Describe your change:

* [ ] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/prime_check.py [hunk 1]
```diff
@@ -32,12 +32,15 @@ def prime_check(number: int) -> bool:
     if 1 < number < 4:
         # 2 and 3 are primes
         return True
-    elif number < 2 or not number % 2:
-        # Negatives, 0, 1 and all even numbers are not primes
+    elif number < 2 or not number % 2 or number % 3 == 0:
+        # Negatives, 0, 1, all even numbers, all multiples of 3 are not primes
```
[the remark this answers is not in the document]
Comment (Lakshmikanth2001) [reply maths/prime_check.py:36]: we can extend this concept to 8  where the prime number will be in form of 8*k +- 1, 8*k +- 3, we need to check with {2, 3, 5, 7} in that case

[the remark this answers is not in the document]
Comment (Lakshmikanth2001) [reply maths/prime_check.py:36]: so 6 is optimal

[the remark this answers is not in the document]
Comment (emelia811) [reply maths/prime_check.py:36]: Nice one

[the remark this answers is not in the document]
Comment (emelia811) [reply maths/prime_check.py:36]: the prime number will be in form of 8k +- 1, 8k +- 3, we need to check with {2, 3, 5, 7} in that case

[the remark this answers is not in the document]
Comment (Lakshmikanth2001) [reply maths/prime_check.py:36]: > we can extend this concept to 8 where the prime number will be in form of 8_k +- 1, 8_k +- 3, we need to check with {2, 3, 5, 7} in that case

yes

[the remark this answers is not in the document]
Comment (emelia811) [reply maths/prime_check.py:36]: 8 option

[the remark this answers is not in the document]
Comment (Lakshmikanth2001) [reply maths/prime_check.py:36]: I didn't get you 😕

File: maths/prime_check.py [hunk 2]
```diff
@@ -32,12 +32,15 @@ def prime_check(number: int) -> bool:
     if 1 < number < 4:
         # 2 and 3 are primes
         return True
-    elif number < 2 or not number % 2:
-        # Negatives, 0, 1 and all even numbers are not primes
+    elif number < 2 or not number % 2 or number % 3 == 0:
```
Comment (poyea) [inline maths/prime_check.py:35]:
```suggestion
    elif number < 2 or number % 2 == 0 or number % 3 == 0:
```

File: maths/prime_check.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (poyea) [reply maths/prime_check.py:36]: I see your point.

File: maths/prime_check.py (hunk 2 above)
Comment (Shubhk32) [reply maths/prime_check.py:35]: What about numbers like 25, 35, 65 etc ?

Comment (Lakshmikanth2001) [reply maths/prime_check.py:35]: Those numbers are divisible by 5

**Comment by Lakshmikanth2001** [comment]
please review my changes and give your valuable feed back

**Comment by poyea** [comment]
Related #6228

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

I'd approve this because this uses one of the properties - I can't tell if it's intuitive or not though

**Pull request merged**